reference/reference.go

```diff
@@ -35,7 +35,7 @@ const (
 
 var (
 	// ErrReferenceInvalidFormat represents an error while trying to parse a string as a reference.
-	ErrReferenceInvalidFormat = errors.New("invalid reference format")
+	ErrReferenceInvalidFormat = errors.New("invalid reference format: repository name must be lowercase")
```

Comment:
Yes, I think it makes sense to return a different error if the reason the parsing fails is because of uppercase characters.